epictrack-api/src/api/reports/thirty_sixty_ninety_report.py

```diff
@@ -159,16 +165,21 @@ def _format_data(self, data):
                 work["work_id"], work["anticipated_decision_date"]
             )
             event_decision_date = work["anticipated_decision_date"]
-            work[
-                "anticipated_decision_date"
-            ] = next_major_decision_event.anticipated_date
+            work["anticipated_decision_date"] = (
+                next_major_decision_event.anticipated_date
+            )
+            project_name = self._get_special_field_project_history(
```

Comment:
This call would go in the for loop. Shall we make it a single call. If we have all the project ids collected prior and make a single call and do the replacement of project name in the works. I think we have the project ids set as index. I am not sure the IN clause would work effectively with indexes though.

Reply:
Updated to do batch queries for projects of 30, 60 & 90 days periods.